### PR TITLE
cleanup: use std::make_shared to replace new

### PR DESCRIPTION
**Pull request by wwformat**
Signed-off-by: Yunchuan Wen <yunchuan.wen@kylin-cloud.com>

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
Looks good at a quick glance. @wwformat, did you use some kind of script to pick these out? Be helpful if you can share it. :)

File: src/ceph_mon.cc [hunk 1]
```diff
@@ -562,7 +562,7 @@ int main(int argc, const char **argv)
     ::encode(v, final);
     ::encode(mapbl, final);
 
-    MonitorDBStore::TransactionRef t(new MonitorDBStore::Transaction);
+    MonitorDBStore::TransactionRef t(std::make_shared<MonitorDBStore::Transaction>());
```
Comment (tchaikov) [inline src/ceph_mon.cc]: nit, why not use `auto` instead?

Comment (wwformat) [inline src/ceph_mon.cc]: OK, I have updated and use auto keyword.

**Comment by tchaikov** [comment]
```
/home/jenkins-build/build/workspace/ceph-pull-requests/qa/workunits/cephtool/test.sh:1842: test_mon_ping:  ceph ping 'mon.*'
...
Error connecting to cluster: TimedOut
...
/home/jenkins-build/build/workspace/ceph-pull-requests/qa/workunits/cephtool/test.sh:1: test_mon_ping:  rm -fr /tmp/cephtool.T95
...

The following tests FAILED:
	  5 - cephtool-test-mon.sh (Failed)
```

retest this please.

**Comment by wwformat** [comment]
> retest this please.

I repeat the test, and the step is:
cd /root/develop/ceph-builder/build
../src/test/cephtool-test-mon.sh

The result is:
/root/develop/ceph-builder/qa/workunits/cephtool/test.sh:2033: main:  echo OK
OK
/root/develop/ceph-builder/qa/workunits/cephtool/test.sh:1: main:  rm -fr /tmp/cephtool.3z4

**Comment by tchaikov** [comment]
@wwformat sorry for the confusion, "retest this please" was a command for jenkins to re-trigger the build/test job. i wanted to see if the failure is transient or not.

**Comment by gregsfortytwo** [comment]
Still wondering how you found/picked out these ones. In the MDS "grep 'new Mutation'" matches the count of changed ones here, but there are several other shared_ptr typedefs which haven't been hit. Are these just the ones you know about and were bugging you? Was there some single script, or set of scripts we'll get another PR on? :)

**Comment by wwformat** [comment]
> Still wondering how you found/picked out these ones. In the MDS "grep 'new Mutation'" matches the count of changed ones here, but there are several other shared_ptr typedefs which haven't been hit. Are these just the ones you know about and were bugging you? Was there some single script, or set of scripts we'll get another PR on? :)

There are many declare of shared_ptr<T> like XXXRef or XXX::Ref.
So, I use a very simple script to filter the code, and pick the right result manullay.
#/bin/bash
git grep -E "^( |\t)*[a-zA-Z0-9_]+(::[a-zA-Z0-9_]+)*( )+[a-zA-Z0-9_]+(::[a-zA-Z0-9_]+)*( |\t)*\(( |\t)*new( |\t)+" | grep Ref
git grep -E "=( )*[a-zA-Z0-9_]+(::[a-zA-Z0-9_]+)*( |\t)*\(( |\t)*new( |\t)+" | grep Ref

**Pull request merged**